desktop_server/src/com/mygdx/game/server/network/EndTurnService.java

```diff
@@ -0,0 +1,68 @@
+package com.mygdx.game.server.network;
+
+import com.artemis.World;
+import com.mygdx.game.core.ecs.component.PlayerToken;
+import com.mygdx.game.server.ecs.entityfactory.ComponentFactory;
+import com.mygdx.game.server.model.Client;
+import com.mygdx.game.server.model.GameRoom;
+import lombok.NonNull;
+import lombok.extern.java.Log;
+
+import javax.inject.Inject;
+import java.util.List;
+
+@Log
+public class EndTurnService {
+
+  private final int playerTokenComponentId;
+  private final PlayerToken playerToken;
+  private final GameRoom gameRoom;
+  private final GameRoomSyncer gameRoomSyncer;
+
+  @Inject
+  EndTurnService(
+      @NonNull World world,
+      @NonNull ComponentFactory componentFactory,
+      @NonNull GameRoom gameRoom,
+      @NonNull GameRoomSyncer gameRoomSyncer
+  ) {
+    this.playerTokenComponentId = componentFactory.createEntityId();
+    this.playerToken = world.getMapper(PlayerToken.class).create(this.playerTokenComponentId);
```

Comment:
Tu tworzysz jeden component, który używasz do każdego gracza, a to źle

Reply:
Zmienione